tools/mqppep/mqppep_preproc.xml

```diff
@@ -0,0 +1,531 @@
+        -n '$networkin'
+        -m '$p_sty_motifs'
+        -p '$psp_kinase_substrate'
+        -r '$psp_regulatory_sites'
+        #if $pst_py_selector == "y"
+            -P y
+        #else
+            -P sty
+        #end if
+        -F $merge_function
+        -o '$mapped_phophopeptides'
+        -O '$melted_phophopeptide_map'
+        -D '$mqppep_output_sqlite'
+        -s '$species'
+      ) && (
+        echo '... end kinase-mapping.'
+      ) && (
+        echo '--- merge-and-filter step:'
+      ) && (
+        python '$__tool_directory__/mqppep_mrgfltr.py'
+        --phosphopeptides='$mapped_phophopeptides'
+        --ppep_mapping_db='$mqppep_output_sqlite'
+        --species='$species'
+        --mrgfltr_tab='$preproc_tab'
+        --mrgfltr_csv='$preproc_csv'
+        --mrgfltr_sqlite='$preproc_sqlite'
+      ) && (
+        echo '... end merge-and-filter.'
+      )
+    ]]></command>
+    <configfiles>
+        <configfile name="phosphocol_script">$phosphoCol
+        </configfile>
+        <configfile name="startcol_script">$startCol
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param name="phosphoSites" type="data" format="tabular"
+               label="Phospho (STY)Sites.txt"
+               help="[phosphoSites] Tabular 'Phospho (STY)Sites.txt' produced by MaxQuant"
+               />
+        <param name="phosphoCol" type="text"
+               label="pattern for column 'Number of Phospho (STY)'"
+               help="[phosphoCol] PERL-compatible regular expression matching header of column having number of 'Phospho (STY)'"
+               value="^Number of Phospho [(]STY[)]$">
+            <sanitizer>
+                <valid initial="string.printable">
+                    <remove value="&apos;"/>
+                </valid>
+            </sanitizer>
+        </param>
+        <param name="startCol" type="text"
+               label="pattern for first column of intensity values"
+               help="[startCol] PERL-compatible regular expression matching column header having first sample intensity"
+               value="^Intensity[^_]">
+            <sanitizer>
+                <valid initial="string.printable">
+                    <remove value="&apos;"/>
+                </valid>
+            </sanitizer>
+        </param>
+        <param name="intervalCol" type="integer" value="1" min="1"
+               label="Interval between the intensity column of samples"
+               help="[intervalCol] E.g., 1 if subsequent column is next sample; 2 if next sample is two columns away, etc."/>
+        <param name="pst_py_selector" type="select"
+            label="Phosphopeptide enrichment type"
+            help="[enrichment] Were samples enriched for pS and pT, or were they enriched for pY instead?"
+            >
+            <option value="st" selected="true">pST</option>
+            <option value="y">pY</option>
+        </param>
+        <param name="collapseFunc" type="select"
```

Comment:
use argument=`--collapse_func`

Reply:
@bgruening Are you saying that I should be matching the param names in the tool.xml with the names of the names of the named arguments passed to the programs being called???

Reply:
changed "collapseFunc" to "collapse_func"